packages/ChangesReporting/ValueObject/RectorWithFileAndLineChange.php

```diff
@@ -41,6 +42,44 @@ public function getRectorClass(): string
         return get_class($this->rector);
     }
 
+    public function getRectorClassWithChangelogUrl(): string
+    {
+        $rectorClass = get_class($this->rector);
+        $changeLogUrl = $this->getChangelogUrl();
+
+        if ($changeLogUrl === null) {
+            return $rectorClass;
+        }
+
+        return sprintf('%s (%s)', $rectorClass, $changeLogUrl);
+    }
+
+    public function getChangelogUrl(): ?string
+    {
+        $rectorClass = get_class($this->rector);
+
+        $rectorReflection = new ReflectionClass($rectorClass);
+
+        $docComment = $rectorReflection->getDocComment();
+
+        if (! is_string($docComment)) {
+            return null;
+        }
+
+        $pattern = "#@link\s*(?<url>[a-zA-Z0-9, ()_].*)#";
+        preg_match($pattern, $docComment, $matches);
+
+        if (! array_key_exists('url', $matches)) {
+            return null;
+        }
+
+        if (! filter_var($matches['url'], FILTER_VALIDATE_URL)) {
+            return null;
+        }
+
+        return trim((string) $matches['url']);
```

Comment:
This seems quite a complext logic for simple value object.
Could you extract this to own service and cover it with simple test?

```php
$extractedValue = $this->extractAnnotationFromClass('SomeRector::class', 'link');
$this->assertSame('...', $extractedValue);
```

Reply:
Where should this service be located? Is there not already something similar we could use?

Reply:
I'll trust your choice :+1: